addons/fire/functions/fnc_burn.sqf

```diff
@@ -0,0 +1,347 @@
+#include "script_component.hpp"
+/*
+ * Author: Brandon (TCVM)
+ * Makes object catch fire. Only call from events. Local effects only.
+ * Arbitrary values to ignite people. Assumed maximum is "10".
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ * 1: Intensity of fire <NUMBER>
+ * 2: Instigator of fire <OBJECT> (default: objNull)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, 4] call ace_fire_fnc_burn
+ *
+ * Public: No
+ */
+
+#define INTENSITY_LOSS 0.03
+#define INTENSITY_UPDATE 3
+#define BURN_PROPOGATE_UPDATE 1
+#define BURN_PROPOGATE_DISTANCE 2
+#define BURN_PROPOGATE_COUNTER_MAX 5
+#define MAX_INTENSITY 10
+#define MIN_INTENSITY 1
+params ["_unit", "_intensity", ["_instigator", objNull]];
+
+if !(GVAR(enabled)) exitWith {};
+
+private _isBurning = [_unit] call FUNC(isBurning);
+if (_isBurning) exitWith {};
+
+[{
+    // looped function
+    (_this getVariable "params") params ["_unit", "", "_instigator"];
+    private _unitPos = getPos _unit;
```

Comment:
If you are on a bridge over water, `getPos` would return `[x,y,0]`
probably better to go with `getPosASL`